sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.CSharp9.cs

```diff
@@ -0,0 +1,28 @@
+﻿record Person
+{
+    private int birthYear;     // Noncompliant {{Make 'birthYear' 'readonly'.}}
+    int birthMonth = 3;        // Noncompliant
+
+    int legSize1 = 3;
+    int legSize2 = 3;
+    bool usedInInit = false;    // FN, only written in property init
+
```

Comment:
what about static fields and `Module Initializers` from C# 9 (which is like a static constructor)?

Reply:
This rule is not relevant for static fields. Only instance fields are inspected.

https://github.com/SonarSource/sonar-dotnet/blob/204f23b90d8c3e70c1921f01df3877989f8ff5f5/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs#L343

Reply:
fair point, the RSPEC mentions `readonly fields can only be assigned in a class constructor. `

I left a comment on the RSPEC anyway https://jira.sonarsource.com/browse/RSPEC-2933

Reply:
although it would be nice to have a test to document this behavior